packages/eslint-plugin/tests/rules/no-useless-template-literals.test.ts

```diff
@@ -0,0 +1,303 @@
+import { noFormat, RuleTester } from '@typescript-eslint/rule-tester';
+
+import rule from '../../src/rules/no-useless-template-literals';
+import { getFixturesRootDir } from '../RuleTester';
+
+const rootPath = getFixturesRootDir();
+
+const ruleTester = new RuleTester({
+  parser: '@typescript-eslint/parser',
+  parserOptions: {
+    tsconfigRootDir: rootPath,
+    project: './tsconfig.json',
+  },
+});
+
+ruleTester.run('no-useless-template-literals', rule, {
+  valid: [
+    "const string = 'a';",
+
+    'const string = `a`;',
+
+    `
+      const string = 'a';
+      const concatenated = \`\${string}b\`;
+    `,
+
+    `
+      const number = 1;
+      const concatenated = \`\${number}b\`;
+    `,
+
+    `
+      const boolean = true;
+      const concatenated = \`\${boolean}b\`;
+    `,
+
+    `
+      const nullish = null;
+      const concatenated = \`\${nullish}-undefined\`;
+    `,
+
+    `
+      const left = 'a';
+      const right = 'b';
+      const concatenated = \`\${left}\${right}\`;
+    `,
+
+    `
+      const left = 'a';
+      const right = 'c';
+      const concatenated = \`\${left}b\${right}\`;
+    `,
+
+    `
+      const left = 'a';
+      const center = 'b';
+      const right = 'c';
+      const concatenated = \`\${left}\${center}\${right}\`;
+    `,
+
+    `
+      const concatenated = \`1 + 1 = \${1 + 1}\`;
+    `,
+
+    `
+      const concatenated = \`true && false = \${true && false}\`;
+    `,
+
+    `
+      tag\`\${'a'}\${'b'}\`;
+    `,
+
+    `
+      const wrappedNumber = \`\${1}\`;
```

Comment:
[Suggestion] Should this be in `invalid`, though? Right now the rule only looks at string types, but I can't think of a reason why it wouldn't flag the useless template literal here. Same with the bigints, booleans, `null`, and `undefined`.

Reply:
hmm,
I wondered the same, but I think this is valid since it has a different meaning than `const value = 1`
it's being converted to a string here, which might be the intention of the developer

Reply:
What I mean is, it's same as `const value = '1'`, no?

Reply:
hmm
so you're saying that in any case where there is any type of literal inside an expression it should be invalid?

Reply:
Yeah that's what I'm thinking. Unless I've missed something -entirely possible, JS coercion/stringification rules are wacky- I think the invalid cases there are unnecessary.

Reply:
yeah, so this is kinda similar to my initial approach, using the `areAllLiterals` if you remember
should be fine (theoretically? I mean.. it's JS after all...)

Reply:
What about things like this?

```TS
`1 + 1 = ${1 + 1}`;
```

Reply:
> initial approach ... `areAllLiterals`

Yeah! I think my concern originally was putting the report range on the whole template literal vs. individual expressions. If an individual expression is any literal, then it should be reported on IMO.

> `${1 + 1}`

Expressions I think we can avoid. We can draw a line at just literals. Anything dynamic, such as a binary expression, can get difficult to report on.

Reply:
Ok, see this one https://github.com/typescript-eslint/typescript-eslint/pull/7957/commits/408e5c721d6e00c7c329d0d355171c67a5fcb2c3